drivers/clock_control/CMakeLists.txt

```diff
@@ -53,9 +53,8 @@ zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex.c)
 
 if(CONFIG_CLOCK_CONTROL_RCAR_CPG_MSSR)
   zephyr_library_sources(clock_control_renesas_cpg_mssr.c)
-if(CONFIG_SOC_R8A77951 OR CONFIG_SOC_R8A7795)
-  zephyr_library_sources(clock_control_r8a7795_cpg_mssr.c)
-endif()
+  zephyr_library_sources_ifdef(CONFIG_DT_HAS_RENESAS_R8A7795_CPG_MSSR_ENABLED clock_control_r8a7795_cpg_mssr.c)
+  zephyr_library_sources_ifdef(CONFIG_SOC_R8A779F clock_control_r8a779f_cpg_mssr.c)
```

Comment:
This looks inconsistent. Why `r8a7795` depends on auto-generated  DTS entry, but  `r8A779f` - on Kconfig option from SOC?